test-data/unit/check-inference.test

```diff
@@ -1585,6 +1585,145 @@ oo.update(d)
 reveal_type(oo) # N: Revealed type is 'collections.OrderedDict[builtins.int*, builtins.str*]'
 [builtins fixtures/dict.pyi]
 
+[case testInferAttributeInitializedToEmptyAndAssigned]
+class C:
+    def __init__(self) -> None:
+        self.a = []
+        if bool():
+            self.a = [1]
```

Comment:
Test using a non-self attribute -- it shouldn't affect type inferece. For example:

```py
...
self.a = []
if bool():
    a = self
    a.a = [1]  # Should not infer type
    a.append(1)  # Should not infer type
```